projects/packages/waf/src/class-waf-blocklog-manager.php

```diff
@@ -0,0 +1,437 @@
+		$date                   = gmdate( 'Y-m-d' );
+		$value                  = intval( $current_value[ $date ] ?? 0 );
+		$current_value[ $date ] = $value + 1;
+
+		return $current_value;
+	}
+
+	/**
+	 * Update the daily summary option in the database.
+	 *
+	 * @param array $value The value to update.
+	 *
+	 * @return void
+	 */
+	private static function write_daily_summary_row( array $value ) {
+		global $table_prefix;
+		$option_name = self::BLOCKLOG_OPTION_NAME_DAILY_SUMMARY;
+
+		$db_connection = self::connect_to_wordpress_db();
+		if ( ! $db_connection ) {
+			return;
+		}
+
+		$updated_value = self::serialize_option_value( $value );
+
+		$statement = $db_connection->prepare( "INSERT INTO {$table_prefix}options (option_name, option_value) VALUES (?, ?) ON DUPLICATE KEY UPDATE option_value = ?" );
+		if ( false !== $statement ) {
+			$statement->bind_param( 'sss', $option_name, $updated_value, $updated_value );
+			$statement->execute();
+		}
+	}
+
+	/**
+	 * Update the daily summary stats for the current date.
+	 *
+	 * @return void
+	 */
+	private static function write_daily_summary() {
+		$stats = self::get_daily_summary();
+		$stats = self::increment_daily_summary( $stats );
+		$stats = self::filter_last_30_days( $stats );
+
+		self::write_daily_summary_row( $stats );
+	}
+
+	/**
+	 * Get the all-time block count value from the database.
+	 *
+	 * @return int The all-time block count.
+	 */
+	private static function get_all_time_block_count_value() {
+		global $table_prefix;
+		$db_connection = self::connect_to_wordpress_db();
+		if ( ! $db_connection ) {
+			return 0;
+		}
+
+		$result = $db_connection->query( "SELECT option_value FROM {$table_prefix}options WHERE option_name = '" . self::BLOCKLOG_OPTION_NAME_ALL_TIME_BLOCK_COUNT . "'" );
+		if ( ! $result ) {
+			return 0;
+		}
+
+		$row = $result->fetch_assoc();
+		if ( ! $row ) {
+			return 0;
+		}
+
+		$all_time_block_count = intval( $row['option_value'] );
+		$result->free();
+
+		return $all_time_block_count;
+	}
+
+	/**
+	 * Update the all-time block count value in the database.
+	 *
+	 * @param int $value The value to update.
+	 * @return void
+	 */
+	private static function write_all_time_block_count_row( int $value ) {
+		global $table_prefix;
+		$option_name = self::BLOCKLOG_OPTION_NAME_ALL_TIME_BLOCK_COUNT;
+
+		$db_connection = self::connect_to_wordpress_db();
+		if ( ! $db_connection ) {
+			return;
+		}
+
+		$statement = $db_connection->prepare( "INSERT INTO {$table_prefix}options (option_name, option_value) VALUES (?, ?) ON DUPLICATE KEY UPDATE option_value = ?" );
+		if ( false !== $statement ) {
+			$statement->bind_param( 'sii', $option_name, $value, $value );
+			$statement->execute();
+		}
+	}
+
+	/**
+	 * Increment the all-time stats.
+	 *
+	 * @return void
+	 */
+	private static function write_all_time_block_count() {
+		$block_count = self::get_all_time_block_count_value() ?? self::get_default_all_time_stat_value();
```

Comment:
```suggestion
		$block_count = self::get_all_time_block_count_value() ?: self::get_default_all_time_stat_value();
```
Whenever the block count is zero, use the default value.